pkg/server/api.go

```diff
@@ -79,13 +79,11 @@ func NewServerAPIHandler(s Server) *APIHandler {
 // API handler.
 func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if r.Method != http.MethodGet && r.Method != http.MethodHead {
-		w.Header().Set("Content-Length", "0")
```

Comment:
Tests expect content length to be 0 in error cases. If we don't encode content length at all, tests need to be adapted.

Reply:
Yup, on it